pkg/pullsync/pullsync.go

```diff
@@ -376,7 +401,7 @@ func (s *Syncer) cursorHandler(ctx context.Context, p p2p.Peer, stream p2p.Strea
 	s.metrics.DbOpsCounter.Inc()
 	ints, err := s.storage.Cursors(ctx)
 	if err != nil {
-		_ = stream.FullClose()
+		_ = stream.Reset()
```

Comment:
No need for reset here as it will be done by deferred function.

Reply:
I think that FullClose was use inappropriate here before, no needf to wait for other side to reset here.